### ISSUE TITLE
Create Chef cookbook for Open PDS

### ISSUE THREAD

**Issue by patcon** [HumanDynamics/openPDS#42]
I'm eager to get a cookbook working to making standing up a server as simple as possible in a secure fashion with server hardening, etc.

Is anyone close to running openpds in a production-ish environment?

**Comment by brian717** [comment]
Good idea. We took many steps in the right direction near the end of 2014 (around the last time you / I / Albert discussed this project) wrt. deployment and documentation on deployment. This made deploying the system more straightforward, but left deployment in a production environment as an exercise for the reader. 

There have been production deployments of openPDS, both in this implementation, as well as other implementations of the SafeAnswers concept. The code itself doesn't typically need to be modified - running each of the components in a separate Docker container and only linking between them as needed is typically sufficient, security-wise. Performance is the bigger issue at scale. 

On that note, I should say that we're planning a bit of a barebones re-write of openPDS with the main goal of making it dead simple to deploy (and simplifying the codebase further by leaving features out of the rewrite that haven't seen much use, or aren't the main focus of the project). Timeframe for starting on this is in the coming weeks, with a working version expected in the coming months. We welcome any contributions you might have.

**Comment by patcon** [comment]
Thanks for the context!  I'll see if I can dig in tonight :)

**Comment by patcon** [comment]
Anyhow, here's the repo if anyone wants to track it, but it's definitely not functional yet:
https://github.com/patcon/chef-openpds

Once I sort out my issues, I'll probably get it working under the [`application_python` cookbook](https://github.com/poise/application_python), get some serverspec tests in place, and then do an official release. Then I'll work on the registry server, if that makes sense.
